internal/uvm/create_wcow.go

```diff
@@ -110,6 +111,11 @@ func CreateWCOW(ctx context.Context, opts *OptionsWCOW) (_ *UtilityVM, err error
 		if err := wcow.CreateUVMScratch(ctx, uvmFolder, scratchFolder, uvm.id); err != nil {
 			return nil, fmt.Errorf("failed to create scratch: %s", err)
 		}
+	} else {
```

Comment:
Can you explain a bit more about this `else` block? I think it handles the case when the a vhdx from the cache is being used?

Reply:
So this is actually something I added to fix the above case in the BUGBUG comment. Right now we always do the UVMs scratch setup here in the shim, but I added work in the snapshotter to do this for us when we know the snapshot is for a WCOW container, so this handles the case where in the snapshot directory "vm/sandbox.vhdx" is already present.

Reply:
So essentially if the work right above this (making the nested vm dir in the snapshot dir and populating it with a sandbox.vhdx) has already been done by the snapshotter, then just grant the UVM access.

Reply:
Do we have any other callers that might call into this method without creation the vm directory and the vhd? I wonder if we can remove the code for creating the directory and vhd (and that comment) now?

Reply:
The BUGBUG comment basically states that it should be up to the caller to do this population so possibly, it would be nice to finally get rid of it haha :) I think we should keep it for now and determine if we can get rid of it moving forward. Even if the containerd changes (E.g. doing this work in the snapshotter itself) doesn't go through, this codes harmless. It just handles the case where some client did the population beforehand.